client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/configure-email-notification/add-new-email-modal.tsx

```diff
@@ -49,36 +90,61 @@ export default function AddNewEmailModal( { toggleModal }: { toggleModal: () =>
 		<Modal
 			open={ true }
 			onRequestClose={ toggleModal }
-			title={ translate( 'Add your email address' ) }
+			title={ title }
 			className="notification-settings__modal"
 		>
-			<div className="notification-settings__sub-title">
-				{ translate( 'Please use only your number or one you have access to. ' ) }
-			</div>
+			<div className="notification-settings__sub-title">{ subTitle }</div>
 
 			<form className="configure-email-notification__form" onSubmit={ onSave }>
 				<FormFieldset>
 					<FormLabel htmlFor="name">{ translate( 'Name' ) }</FormLabel>
-					<FormTextInput id="name" name="name" disabled={ showCodeVerification } />
-					<div className="configure-email-notification__help-text">
-						{ translate( 'Give this email a nickname for your personal reference' ) }
-					</div>
+					<FormTextInput
+						id="name"
+						name="name"
+						value={ emailItem.name }
+						disabled={ showCodeVerification }
+						onChange={ ( event: { target: { value: string } } ) =>
```

Comment:
We can maybe `useCallback`  and avoid rerendering the rest of the text input fields when we update one of the values (email, name, or code).